File: rust/src/api/ingest.rs
// this file is @generated
use super::{IngestAuthentication, IngestEndpoint, IngestSource};
use crate::Configuration;

pub struct Ingest<'a> {
    cfg: &'a Configuration,
}

impl<'a> Ingest<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn authentication(&self) -> IngestAuthentication<'a> {
        IngestAuthentication::new(self.cfg)
    }

    pub fn endpoint(&self) -> IngestEndpoint<'a> {
        IngestEndpoint::new(self.cfg)
    }

    pub fn source(&self) -> IngestSource<'a> {
        IngestSource::new(self.cfg)
    }
}

File: javascript/src/api/ingest.ts
// this file is @generated

import { IngestAuthentication } from "./ingestAuthentication";
import { IngestEndpoint } from "./ingestEndpoint";
import { IngestSource } from "./ingestSource";
import type { SvixRequestContext } from "../request";

export class Ingest {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  public get authentication() {
    return new IngestAuthentication(this.requestCtx);
  }

  public get endpoint() {
    return new IngestEndpoint(this.requestCtx);
  }

  public get source() {
    return new IngestSource(this.requestCtx);
  }
}
